corehq/apps/hqwebapp/static/hqwebapp/js/ajax_csrf_setup.js

```diff
@@ -0,0 +1,13 @@
+function csrfSafeMethod(method) {
+    // these HTTP methods do not require CSRF protection
+    return (/^(GET|HEAD|OPTIONS|TRACE)$/.test(method));
+}
+
+function setAjaxCsrfHeader(xhr, settings) {
+	$csrf_token = $.cookie('csrftoken');
```

Comment:
could put this inside the if so that it doesn't have to load the cookie unless necessary